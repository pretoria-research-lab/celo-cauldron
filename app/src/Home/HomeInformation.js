import React from 'react';

export default function HomeInformation(props) {

    return (
        <div id="home-info" className="row">
          <div className="col-lg-12">
              <p>A repository of useful tools, scripts, and information for developing on the Celo ecosystem.</p>
          </div>
        </div>
    );
}