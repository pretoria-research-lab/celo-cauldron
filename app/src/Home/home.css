.celoIcons {
    max-width: 100px;
}

.tools {
    align-items: center;
}

.link-list {    
    background-color: var(--celo-canvas-faint-gold);   
    border: solid 1px var(--celo-accent-cyan);
    padding: 1%;
    margin-top: 10px;
}

.jost {
 font-family: Jost;
}

ul {
    text-align: left;
    font-size: larger;
}

#left {
 text-align: right;
 padding-left: 5px;
}

td {
    padding-right: 10px;
}

#right {
    text-align: left;
    padding-right: 5px;
}

.small-padding {
    padding-bottom: 0px;
}