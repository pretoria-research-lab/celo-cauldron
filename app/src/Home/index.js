import React from "react";
import "./home.css";
import HomeInformation from "./HomeInformation";
import "bootstrap/dist/css/bootstrap.css";
import "bootstrap";
import doersLogo from "../assets/04-Doers (light bg).png";

export default function Home() {
	return (
		<div className="container-fluid">    
			<div className="column centered">
				<div className="row page-header">      
					<img className="celoIcons" src={doersLogo} alt="Doers icon" /> 
					<h1 className="mt">Celo Cauldron</h1>					
				</div>
				<div className="row">
					<h4>An independent community list of useful tools, scripts, and information for developing on the Celo ecosystem.</h4>
				</div>
			</div>
			
			<hr/>
			<HomeInformation />
			<hr/>
		</div>
	);
}
