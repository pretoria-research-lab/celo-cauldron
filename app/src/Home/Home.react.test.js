import React from 'react';
import Home from './index';
import renderer from 'react-test-renderer';

const faucets = ["Alfajores", "Baklava", "RC 1", "Mainnet"];

test('Home component matches snapshot', () => {
    const component = renderer.create(
      <Home faucets={faucets}/>,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot(); 
  });