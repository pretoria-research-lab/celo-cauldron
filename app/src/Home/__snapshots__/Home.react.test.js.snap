// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="column centered"
  >
    <div
      className="row page-header"
    >
      <img
        alt="Doers icon"
        className="celoIcons"
        src="04-Doers (light bg).png"
      />
      <h1
        className="mt"
      >
        Cauldron
      </h1>
    </div>
  </div>
  <hr />
  <div
    className="row"
  >
    <div
      className="jost col-lg-12"
    >
      <h3>
        A repository of useful tools, scripts, and information for developing on the Celo ecosystem.
      </h3>
    </div>
  </div>
  <div
    className="column centered"
  >
    <div
      className="link-list col-lg-8"
    >
      <h2>
        <img
          alt="cLabs resource"
          className="tiny-icon"
          src="11-Security Check (light-bg).png"
        />
         cLabs
      </h2>
      <hr />
      <table
        className="responsive col-lg-12"
      >
        <tbody>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Alfajores Network Statistics
            </td>
            <td
              id="right"
            >
              <a
                href="https://alfajores-celostats.celo-testnet.org/"
              >
                https://alfajores-celostats.celo-testnet.org/
              </a>
            </td>
          </tr>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Alfajores Testnet block explorer
            </td>
            <td
              id="right"
            >
              <a
                href="https://alfajores-blockscout.celo-testnet.org/"
              >
                https://alfajores-blockscout.celo-testnet.org/
              </a>
            </td>
          </tr>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Baklava Network Statistics
            </td>
            <td
              id="right"
            >
              <a
                href="https://baklava-celostats.celo-testnet.org/"
              >
                https://baklava-celostats.celo-testnet.org/
              </a>
            </td>
          </tr>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Baklava Testnet block explorer
            </td>
            <td
              id="right"
            >
              <a
                href="https://baklava-blockscout.celo-testnet.org/"
              >
                https://baklava-blockscout.celo-testnet.org/
              </a>
            </td>
          </tr>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Celo Documentation
            </td>
            <td
              id="right"
            >
              <a
                href="https://docs.celo.org/"
              >
                https://docs.celo.org/
              </a>
            </td>
          </tr>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Hackerone Celo Bug Bounty Program
            </td>
            <td
              id="right"
            >
              <a
                href="https://hackerone.com/celo/"
              >
                https://hackerone.com/celo/
              </a>
            </td>
          </tr>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Mentorship and Funding @ Celo Camp
            </td>
            <td
              id="right"
            >
              <a
                href="https://www.celocamp.com/"
              >
                https://www.celocamp.com/
              </a>
            </td>
          </tr>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Validator Rewards Calculator
            </td>
            <td
              id="right"
            >
              <a
                href="https://observablehq.com/@celohq/validator-rewards-calculator"
              >
                https://observablehq.com/@celohq/validator-rewards-calculator
              </a>
            </td>
          </tr>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              cLabs Alfajores Faucet
            </td>
            <td
              id="right"
            >
              <a
                href="https://celo.org/build/faucet"
              >
                https://celo.org/build/faucet
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="link-list col-lg-8"
    >
      <h2>
        <img
          alt="Community based resource"
          className="tiny-icon"
          src="03-People (light-bg).png"
        />
         Community
      </h2>
      <hr />
      <table
        className="responsive col-lg-12"
      >
        <tbody>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Baklava Validator Group Topography
            </td>
            <td
              id="right"
            >
              <a
                href="https://celomap.io"
              >
                https://celomap.io
              </a>
            </td>
          </tr>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Baklava Validator Statistics
            </td>
            <td
              id="right"
            >
              <a
                href="https://thecelo.com"
              >
                https://thecelo.com
              </a>
            </td>
          </tr>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Celo Maintenance Scripts
            </td>
            <td
              id="right"
            >
              <a
                href="https://gist.github.com/yourcodesucks/374c30454ec1d93af028c153baae3bed"
              >
                https://gist.github.com/yourcodesucks/374c30454ec1d93af028c153baae3bed
              </a>
            </td>
          </tr>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Celo Node DevOps Tooling and Infrastructure
            </td>
            <td
              id="right"
            >
              <a
                href="https://github.com/jim380/node_tooling/tree/master/Celo"
              >
                https://github.com/jim380/node_tooling/tree/master/Celo
              </a>
            </td>
          </tr>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Celo Validator Dashboard
            </td>
            <td
              id="right"
            >
              <a
                href="https://celowhale.com"
              >
                https://celowhale.com
              </a>
            </td>
          </tr>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Celo Validator Monitor Discord Bot
            </td>
            <td
              id="right"
            >
              <a
                href="https://github.com/dsrvlabs/celo-validator-monitoring"
              >
                https://github.com/dsrvlabs/celo-validator-monitoring
              </a>
            </td>
          </tr>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Community Alfajores Faucet
            </td>
            <td
              id="right"
            >
              <a
                href="https://cauldron.pretoria.tech/alfajores-faucet"
              >
                https://cauldron.pretoria.tech/alfajores-faucet
              </a>
            </td>
          </tr>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Community Baklava Faucet
            </td>
            <td
              id="right"
            >
              <a
                href="https://cauldron.pretoria.tech/baklava-faucet"
              >
                https://cauldron.pretoria.tech/baklava-faucet
              </a>
            </td>
          </tr>
          <tr
            className="small-padding"
          >
            <td
              id="left"
            >
              Project and Concepts Mindmap
            </td>
            <td
              id="right"
            >
              <a
                href="https://thecelo.com/celo.html"
              >
                https://thecelo.com/celo.html
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <hr />
</div>
`;
